src/dotnet/CarbonAware.DataSources.WattTime/test/WattTimeDataSourceTests.cs

```diff
@@ -118,11 +118,10 @@ public void GetCarbonIntensity_ThrowsWhenRegionNotFound()
     private void SetupBalancingAuthority(BalancingAuthority balancingAuthority, Location location)
     {
         this.LocationSource.Setup(r => r.ToGeopositionLocationAsync(location)).Returns(Task.FromResult(location));
-        
-        this.WattTimeClient.Setup(w => w.GetBalancingAuthorityAsync(
-            location.Latitude.ToString() ?? "",
-            location.Longitude.ToString() ?? ""
-            )
+        var latitude = location.Latitude.ToString() ?? throw new Exception();
```

Comment:
This should be slightly more specific
```suggestion
        var latitude = location.Latitude.ToString() ?? throw new ArgumentNullException(String.Format("Could not find location"));
```